CMakeLists.txt

```diff
@@ -40,9 +34,17 @@ ament_target_dependencies(${PROJECT_NAME}_node
   sensor_msgs
   std_msgs
   tf2_ros
-  urdf
-)
+  urdf)
+
+add_library(
+  ${PROJECT_NAME}_node SHARED
+  src/robot_state_publisher.cpp)
+target_include_directories(${PROJECT_NAME}_node PUBLIC
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")
+ament_target_dependencies(${PROJECT_NAME}_node ${THIS_PACKAGE_INCLUDE_DEPENDS})
 ament_export_targets(export_${PROJECT_NAME}_node)
+ament_export_dependencies(${THIS_PACKAGE_INCLUDE_DEPENDS})
```

Comment:
Note that we already have an `ament_export_dependencies` down below (on line 117).  So we shouldn't need to add another one.  That said, that line is missing some of the dependencies (most notably `geometry_msgs`, `kdl_parser`, `rcl_interfaces`, and `rclcpp_components`).  So I'd suggest that we remove this newly added line here, and instead change line 117 to be:

```
ament_export_dependencies(${THIS_PACKAGE_INCLUDE_DEPENDS})
```